HiP-FeatureToggle/Controllers/FeaturesController.cs

```diff
@@ -0,0 +1,163 @@
+    [Route("Api/[controller]")]
+    public class FeaturesController : Controller
+    {
+        private readonly FeaturesManager _manager;
+        private readonly CmsService _cmsService;
+
+        private bool IsAdministrator => _cmsService.GetUserRole(User.Identity.GetUserIdentity()) == "Administrator";
+
+        public FeaturesController(FeaturesManager manager, CmsService cmsService)
+        {
+            _manager = manager;
+            _cmsService = cmsService;
+        }
+
+        /// <summary>
+        /// Gets all features.
+        /// </summary>
+        [HttpGet]
+        [ProducesResponseType(typeof(IEnumerable<FeatureResult>), 200)]
+        [ProducesResponseType(403)]
+        public IActionResult GetAll()
+        {
+            if (!IsAdministrator)
+                return Forbid();
+
+            var features = _manager.GetFeatures(loadChildren: true, loadGroups: true);
+            var results = features.ToList().Select(f => new FeatureResult(f));
+            return Ok(results);
+        }
+
+        /// <summary>
+        /// Gets a specific feature by ID.
+        /// </summary>
+        [HttpGet("{featureId}")]
+        [ProducesResponseType(typeof(FeatureResult), 200)]
+        [ProducesResponseType(403)]
+        public IActionResult GetById(int featureId)
+        {
+            if (!IsAdministrator)
+                return Forbid();
+
+            var feature = _manager.GetFeature(featureId, loadChildren: true, loadGroups: true);
+
+            if (feature == null)
+                return NotFound();
+
+            return Ok(new FeatureResult(feature));
+        }
+
+        /// <summary>
+        /// Stores a new feature.
+        /// </summary>
+        [HttpPost]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(403)]
+        [ProducesResponseType(409)]
+        public IActionResult Create([FromBody]FeatureArgs args)
+        {
+            if (!IsAdministrator)
+                return Forbid();
+
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            try
+            {
+                _manager.CreateFeature(args);
+                return Ok();
+            }
+            catch (ResourceNotFoundException<Feature> e)
+            {
+                return StatusCode(422, e.Message); // invalid parent feature ID
+            }
+            catch (ArgumentException e)
+            {
+                return StatusCode(409, e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Deletes a feature. If the feature has children, these are made children of the deleted feature's parent.
+        /// </summary>
+        /// <param name="featureId"></param>
+        /// <returns></returns>
+        [HttpDelete("{featureId}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(403)]
+        [ProducesResponseType(404)]
+        public IActionResult Delete(int featureId)
```

Comment:
I was able to delete a feature that was the parent of other features -- could be problematic as this destroyes the whole hierarchy (the chidl features were still existing after deleting the parent, with the `parent` attribute set to null)

My advice would be not allowing deletion of a parent feature at all; alternatively you could also delete all the child features